smac_planner/src/smac_planner.cpp

```diff
@@ -239,22 +259,30 @@ nav_msgs::msg::Path SmacPlanner::createPlan(
   steady_clock::time_point a = steady_clock::now();
 #endif
 
+  // Choose which costmap to use for the planning
+  nav2_costmap_2d::Costmap2D * costmap;
+  if (_costmap_downsampler) {
+    costmap = _costmap_downsampler->downsample();
+  } else {
```

Comment:
Remove the else statement, we can just have an `if()` to change the pointer address, the else case can just use the initialized `_costmap`.